### DISCUSSION TITLE
Next.js 15 RC pages with fetch still static

### DISCUSSION THREAD

**Discussion by ghocevar** [vercel/next.js#66209] [Help]
### Summary

I have a question regarding pages with fetch in Next.js 15. In version 15 all fetch call shouldn't be cached by default (if not opt-in). Shouldn't a page where a fetch call is made, be dynamic by default?

You can see same issue in [Jack Herrington's video](https://youtu.be/N2LzvfM2R5M?si=su8VgktZHs6MYtcZ&t=775).

<img width="864" alt="image" src="https://github.com/vercel/next.js/assets/51193034/c16b31ca-287e-4928-99cc-8643f55052ee">

You can see that `next build` outputs `/` page as static and not dynamic. If you uncomment `export const dynamic = 'force-dynamic'` or add `noStore()` into `getTime()` function, page becomes dynamic.

### Additional information

```js
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.5.0: Wed May  1 20:12:58 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6000
  Available memory (MB): 32768
  Available CPU cores: 10
Binaries:
  Node: 20.12.2
  npm: 10.5.0
  Yarn: 1.22.21
  pnpm: 9.1.1
Relevant Packages:
  next: 15.0.0-rc.0 // Latest available version is detected (15.0.0-rc.0).
  eslint-config-next: 15.0.0-rc.0
  react: 19.0.0-rc-f994737d14-20240522
  react-dom: 19.0.0-rc-f994737d14-20240522
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

### Example

https://github.com/ghocevar/nextjs-15-0-rc

**Comment by SidKH** [comment]
> Shouldn't a page where a fetch call is made, be dynamic by default?

This could make it more confusing.

```javascript
// Will make page dynamic
await fetch(...);

// Won't make page dynamic
await db.item.findUnique({ id })
```


It seems like the underlying issue is that the choice of rendering **is not explicit**. Changing opt-in/opt-out conditions would hardly help. 

Plus, the final strategy should lead towards PPR, so most pages should be static anyway.

---

The question is - whether it's possible to make the static/dynamic rendering decision more explicit somehow?

**Reply by AkifumiSato** [reply]
The documentation also states that `no-store` is the default.
https://rc.nextjs.org/docs/app/api-reference/functions/fetch#optionscache

However, as far as I have tried, it doesn't seem to be dynamic rendering unless explicitly specified `no-store`.
```ts
// Won't make page dynamic
await fetch("https://dummyjson.com/todos/random")

// Will make page dynamic
await fetch("https://dummyjson.com/todos/random", {
  cache: "no-store",
})
```

---

Dynamic seems more natural to me. Wouldn't it be better to at least have a description in the documentation?

**Reply by AkifumiSato** [reply]
ref: #66240

**Comment by SidKH** [comment]
Also, any additional indication of whether the current page is static or dynamic will help immensely. 

![CleanShot 2024-05-26 at 12 51 47@2x](https://github.com/vercel/next.js/assets/7762253/9be11bf3-d779-4dea-b67e-c0f2197d5c8d)

**Comment by ghocevar** [comment]
The same is happening on [v15.0.0-canary.1 release](https://github.com/vercel/next.js/releases/tag/v15.0.0-canary.1). 

cc: @leerob @delbaoliveira
Could you please clarify this behavior? Thank you. ✌️

**Reply by delbaoliveira** [reply]
Next.js 15 RC changes the default behavior of `fetch` but not rendering, which is still static unless you use a [dynamic API](https://nextjs.org/docs/app/building-your-application/rendering/server-components#dynamic-functions). 

So you'll notice the change for dynamic routes which had `fetch` requests that were cached, but as of today, an uncached fetch will not opt a page into dynamic rendering. You can quickly opt into dynamic rendering using [noStore()](https://nextjs.org/docs/app/api-reference/functions/unstable_noStore). 

We're still working on the default for rendering for Next.js 15 GA, more 🔜 .

**Reply by DevDynasty0** [reply]
Thank @delbaoliveira for this clarification. I struggled to comprehend this for a long time.


But in the Next.js [docs](https://nextjs.org/docs/app/building-your-application/rendering/server-components#switching-to-dynamic-rendering) it is said that, "During rendering, if a [Dynamic API](https://nextjs.org/docs/app/building-your-application/rendering/server-components#dynamic-apis) or `uncached data request` is discovered, Next.js will switch to dynamically rendering the whole route." 
It creates confusion because isn't the fetch request in Next.js 15 an `uncached data request`?

**Reply by mauriciotp** [reply]
I have the same question, it's very confusing in the documentation

**Reply by a89529294** [reply]
[here](https://nextjs.org/docs/app/api-reference/functions/fetch#optionscache) in the docs it says "auto no cache (default): Next.js fetches the resource from the remote server on every request in development, but will fetch once during next build because the route will be statically prerendered." 
After I ran `npm run build` I still see in .next/cache/fetch-cache
```js
{
"url":"http://www.randomnumberapi.com/api/v1.0/random?min=100&max=1000&count=1"},"revalidate":31536000,"tags":[]}
```
This is a simple app with just a fetch call to 'http://www.randomnumberapi.com/api/v1.0/random?min=100&max=1000&count=1' in the root page.tsx. Shouldn't the "revalidate" key be set to 0? It seems like it is still cached.
After I rebuild it, the number returned still stays the same, which suggests the fetch call was indeed cached.
